sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_fixed_point.asciidoc

```diff
@@ -0,0 +1,501 @@
+
+----
+OpFixedSqrtINTEL
+OpFixedRecipINTEL
+OpFixedRsqrtINTEL
+OpFixedSinINTEL
+OpFixedCosINTEL
+OpFixedSinCosINTEL
+OpFixedSinPiINTEL
+OpFixedCosPiINTEL
+OpFixedSinCosPiINTEL
+OpFixedLogINTEL
+OpFixedExpINTEL
+----
+
+== Token Number Assignments
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|`ArbitraryPrecisionFixedPointINTEL`    | 5922
+|`OpFixedSqrtINTEL`                     | 5923
+|`OpFixedRecipINTEL`                    | 5924
+|`OpFixedRsqrtINTEL`                    | 5925
+|`OpFixedSinINTEL`                      | 5926
+|`OpFixedCosINTEL`                      | 5927
+|`OpFixedSinCosINTEL`                   | 5928
+|`OpFixedSinPiINTEL`                    | 5929
+|`OpFixedCosPiINTEL`                    | 5930
+|`OpFixedSinCosPiINTEL`                 | 5931
+|`OpFixedLogINTEL`                      | 5932
+|`OpFixedExpINTEL`                      | 5933
+|====
+
+
+== Modifications to the SPIR-V Specification Version 1.5
+
+After Section 3.16, add a new section "3.16b Quantization Modes" as follows:
+
+=== Quantization Modes
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode        ^| Behavior
+| 0      | TRN          | Truncate towards -Inf
+| 1      | TRN_ZERO     | Truncate towards 0
+| 2      | RND          | Round towards +Inf
+| 3      | RND_ZERO     | Round towards 0
+| 4      | RND_INF      | Round towards +/-Inf
+| 5      | RND_MIN_INF  | Round towards -Inf
+| 6      | RND_CONV     | Round towards even
+| 7      | RND_CONV_ODD | Round towards odd
+|====
+
+After Section 3.16, add a new section "3.16c Overflow Modes" as follows:
+
+=== Overflow Modes
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode     ^| Behavior
+| 0      | WRAP      | Drop the bits to the left of the MSB
+| 1      | SAT       | Saturate to the closest of MIN or MAX
+| 2      | SAT_ZERO  | Set to 0 on overflow
+| 3      | SAT_SYM   | For unsigned, treat as SAT. +
+
+For signed: a positive overflow will saturate at the maximum positive value, whereas a negative overflow will saturate to the negation of the maximum positive value, as opposed to the most negative value.
+|====
+
+=== Capability
+Modify Section 3.31, *Capability*, adding a row to the Capability table:
+
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 5922 | *ArbitraryPrecisionFixedPointINTEL* +
+
+Enables arbitrary precision fixed-point math instructions
+| *INTELArbitraryPrecisionIntegers*
+|====
+
+=== Instructions
+
+In Section 3.32.13, *Arithmetic Instructions*, add the following instructions:
+
+[cols="11", width="100%"]
+|=====
+10+<|*OpFixedSqrtINTEL* +
+
+An `OpTypeInt` representing an arbitrary precision fixed point number (ac_fixed) is passed in as the _Input_ and the square root of the value is returned in _Result_.
+The behavior of this function is undefined for input values < 0.
+
+_Result Type_ is an `OpTypeInt` of width _rW_ and is the type of _Result_.
```

Comment:
We don't have plans to map this for OpenCL, just for SYCL, and we plan on doing the data validation statically in the C++ header file that leverages these built-ins. To avoid running validation checks in both places, I omitted them from here.